src/codegen.cpp

```diff
@@ -7559,10 +7672,16 @@ static jl_llvm_functions_t
     // Step 4b. determine debug info signature and other type info for locals
     DICompileUnit::DebugEmissionKind emissionKind = (DICompileUnit::DebugEmissionKind) ctx.params->debug_info_kind;
     DICompileUnit::DebugNameTableKind tableKind;
+#if JL_LLVM_VERSION >= 160000
+#undef None
```

Comment:
IMO we should just define `template<typename T> using Optional = std::optional<T>` and replace the explicit function calls with the ones supported by both implementations (e.g. operator bool, operator*, etc). That'll reduce the diff size significantly.